lib/backend/syncersv1/updateprocessors/rules.go

```diff
@@ -140,20 +165,62 @@ func RuleAPIV2ToBackend(ar apiv3.Rule, ns string) model.Rule {
 	}
 }
 
-// parseNamespaceSelector takes a v3 namespace selector and returns the appropriate v1 representation
-// by prefixing the keys with the `pcns.` prefix. For example, `k == 'v'` becomes `pcns.k == 'v'`.
-func parseNamespaceSelector(s string) string {
+// parseSelectorAttachPrefix takes a v3 selector and returns the appropriate v1 representation
+// by prefixing the keys with the given prefix.
+// If prefix is `pcns.` then the selector changes from `k == 'v'` to `pcns.k == 'v'`.
+func parseSelectorAttachPrefix(s, prefix string) string {
 	parsedSelector, err := parser.Parse(s)
 	if err != nil {
-		log.WithError(err).Errorf("Failed to parse namespace selector: %s", s)
+		log.WithError(err).Errorf("Failed to parse selector: %s (for prefix) %s", s, prefix)
 		return ""
 	}
-	parsedSelector.AcceptVisitor(parser.PrefixVisitor{Prefix: conversion.NamespaceLabelPrefix})
+	parsedSelector.AcceptVisitor(parser.PrefixVisitor{Prefix: prefix})
 	updated := parsedSelector.String()
-	log.WithFields(log.Fields{"original": s, "updated": updated}).Debug("Updated namespace selector")
+	log.WithFields(log.Fields{"original": s, "updated": updated}).Debug("Updated selector")
 	return updated
 }
 
+// parseServiceAccounts takes a v3 service account match and returns the appropriate v1 representation
+// by converting the list of service account names into a set of service account with
+// key: "projectcalico.org/serviceaccount" in { 'sa-1', 'sa-2' } AND
+// by prefixing the keys with the `pcsa.` prefix. For example, `k == 'v'` becomes `pcsa.k == 'v'`.
+// NOTE: Check if the env variable ALPHA_FEATURES is set or not before converting the ServiceAccountMatch to selector
+func parseServiceAccounts(sam *apiv3.ServiceAccountMatch) string {
+
+	if apiconfig.IsAlphaFeatureSet(os.Getenv("ALPHA_FEATURES"), apiconfig.AlphaFeatureSA) == false {
```

Comment:
Shouldn't this use the client apiconfig rather than the environment?  What if a user sets this in the config file but not in the environment - this check will fail.

Reply:
yes. ideally! getting apiconfig spec all the way down to updateprocessors was looking a bit hairy.
In talking with @caseydavenport for the interim we came up with using the env. This is going to get invoked in caliconode and there we do set the `ALPHA_FEATURES`
The plan is to make the required changes to make that happen post-merge of this PR.